aten/src/ATen/cuda/CUDAGdsFile.cpp

```diff
@@ -0,0 +1,177 @@
+#include <ATen/cuda/CUDAGdsFile.h>
+
+// torch
+#include <c10/cuda/CUDAGuard.h>
+
+// cuda
+#include <cuda_runtime.h>
+#include <cufile.h>
+#include <chrono>
+#include <iostream>
+
+// file io
+#include <unistd.h>
+#include <fcntl.h>
+
+
+namespace at::cuda {
```

Comment:
This would be a lot simpler if this whole logic was directly in torch/csrc/cuda/GdsFile.cpp
If you have any issue with this new aten file, I would suggest just moving the logic there. If a c++-only user needs this API, they can already just use cuda's API directly, so I don't think we need to expose it as a pure c++ API on atne.